src/main/java/frc/trigon/robot/subsystems/arm/toohardarm/ToohardArmIO.java

```diff
@@ -0,0 +1,109 @@
+    private final CANSparkMax
+            masterAngleMotor = ToohardArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ToohardArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ToohardArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ToohardArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final TalonSRX elevatorEncoder = ToohardArmConstants.ELEVATOR_ENCODER;
+
+    @Override
+    protected void updateInputs(ArmInputsAutoLogged inputs) {
+        inputs.elevatorMotorVoltage = masterElevatorMotor.getBusVoltage();
+        inputs.elevatorPositionMeters = getElevatorPositionMeters();
+        inputs.elevatorVelocityMetersPerSecond = getElevatorVelocityMetersPerSecond();
+
+        inputs.angleMotorVoltage = masterAngleMotor.getBusVoltage();
+        inputs.anglePositionDegrees = getAnglePositionDegrees();
+        inputs.angleVelocityDegreesPerSecond = getAngleVelocityDegreesPerSecond();
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleMotorsVoltage(calculateAngleVoltageFromState(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorMotorsVoltage(calculateElevatorVoltageFromState(targetState));
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        masterAngleMotor.stopMotor();
+        followerAngleMotor.stopMotor();
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        masterElevatorMotor.stopMotor();
+        followerElevatorMotor.stopMotor();
+    }
+
+    private double calculateElevatorVoltageFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = ToohardArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorPositionMeters(),
+                targetState.position
+        );
+        double feedforward = ToohardArmConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        double voltage = Conversions.compensatedPowerToVoltage(pidOutput + feedforward, ToohardArmConstants.VOLTAGE_COMPENSATION_SATURATION);
+        return MathUtil.clamp(
+                voltage,
+                -ToohardArmConstants.VOLTAGE_COMPENSATION_SATURATION,
+                ToohardArmConstants.VOLTAGE_COMPENSATION_SATURATION
+        );
+    }
+
+    private double calculateAngleVoltageFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = ToohardArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAnglePositionDegrees(),
+                targetState.position
+        );
+        double feedforward = ToohardArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                targetState.velocity
+        );
+        double voltage = Conversions.compensatedPowerToVoltage(pidOutput + feedforward, ToohardArmConstants.VOLTAGE_COMPENSATION_SATURATION);
```

Comment:
pid and ff return an output. Based on how they're tuned, this could already return voltage based on the numbers, so no need for this.